### ISSUE TITLE
Setting trust

### ISSUE THREAD

**Issue by agaida** [lxqt/libfm-qt#354]
##### Expected Behavior
Should be indipendent of the pcmanfm-qt settings - Rationale: if we override it, we should be able to set the trust to a desktopfile -  the untrusted emblem looks fugly :) 

##### Current Behavior
can set the trust only if global is off

##### Possible Solution
change the implementation logic

current release

**Comment by agaida** [comment]
god damn - set the global thing - if you have untrusted desktop files, try to trust them - you will not be able to do so, because there is no context switch anymore :smile:

**Comment by agaida** [comment]
My dear friend Donald would likely describe it as: Sad, so sad ...

**Comment by tsujan** [comment]
> god damn - set the global thing

Would you please tell me what you mean by "global thing"; it's like Klingon language...

> if you have untrusted desktop files, try to trust them  - you will not be able to do so, because there is no context switch anymore

Of course, I *can* trust them; tried it a few seconds ago!

**Comment by tsujan** [comment]
@agaida If this issue was opened by a user, I'm sure you'd close it immediately.

**Comment by tsujan** [comment]
A screenshot:

![trust](https://user-images.githubusercontent.com/981076/51871903-ed870600-236c-11e9-8225-f51b0a332d2f.png)

**Comment by agaida** [comment]
![](https://i.imgur.com/QgPDjfr.png) - and it was a bug reported by a trustworth user

**Comment by tsujan** [comment]
Now, it's in English. It is a minor issue *only on Desktop*. Checking that checkbox doesn't and shouldn't add the "trust" metadata to all desktop files; it just launches them -- user's responsibility.

If you open the same file from pcmanfm-qt's window. it'll be launched without prompt. I was aware that it wouldn't on Desktop itself but simply forgot it among so many jobs... because it wasn't important.

Moreover, I think many people mistake "desktop files" with files on desktop ;)

**Comment by tsujan** [comment]
And wait a minute: THERE IS NOTHING TO FIX. Do you know why? It says, "Applications restart is needed..."

Please close this issue immediately or I'll get angry...

**Comment by tsujan** [comment]
And trust me more than trustworthy users.

**Comment by agaida** [comment]
Nope, wouldn't work in this case. And i reproduce things before filing bugs - maybe this picture make it more clear.
![](https://i.imgur.com/HYMrJOW.jpg)

I can take a picture with the hook unset if it isn't obvious enough.

**Comment by agaida** [comment]
here it is:

![](https://i.imgur.com/XLXbMLt.jpg)

**Comment by agaida** [comment]
The content menu entry should be visible indipendent from the settings in pcmanfm-qt - some people don't like the exclamation mark emblem - unless they want to disable the prompt.

**Comment by tsujan** [comment]
It's a reasonable, correct and intended behavior. A user that checks that option shouldn't be able to change that metadata but he/she should see what he/she launches carelessly.

Let me guess -- a little mind reading: That "trustworthy" user first complained about the emblem, as if he'd never seen it elsewhere (perhaps he has an ugly icon set), and then wanted to get rid of it by something totally unrelated. There's no trace of the word "trust" in that option.

**Comment by tsujan** [comment]
Never! I don't want to see the security complaints again. They were right, I agreed with the users that mentioned them and added this mechanism.

I think you're caught in a dubious discussion with that unknown user. There are lots of other users.

**Comment by tsujan** [comment]
A computer user should be able to right click on multiple files after toggling that option (how many desktop files can one have? it's not important in this case) and trust all of them at once. There is no problem to solve -- just an odd obsession.

**Comment by agaida** [comment]
This has nothing to do with security, this has only to do with a damn moronic workflow. How to get rid of the emblems when the user decide not to prompt? Enable prompting, change the files, disable prompting? 

NO, not really, this can't be meant serious.

**Comment by tsujan** [comment]
> How to get rid of the emblems when the user decide not to prompt?

I explained above, although I knew you wouldn't need it under usual circumstances. Moreover, those emblems aren't a plague to get rid of (hence the word "obsession").

Disabling prompt = Launching carelessly ≠ Trusting = Security.

**Comment by tsujan** [comment]
I know what I'm saying because I listened to users and did it for them -- and for myself too because they convinced me.

**Comment by tsujan** [comment]
Knowing that I might be more accused of mixing things, I can't ignore the similarity between this and https://github.com/lxqt/lxqt/issues/1661 -- they're identical, in a sense.

**Comment by agaida** [comment]
no - but i'm dead sure that this don't go into debian stable - congrats, i have to patch this out. I can't set my name under.

**Comment by tsujan** [comment]
I removed the assignment because I won't undo a work of mine that I consider correct.

**Comment by agaida** [comment]
do what you want to do - i take a vacation for a undefined time

**Comment by palinek** [comment]
After reading these all, the only needed thing is to allow the trust/untrust in the context menu for .desktop files even if the the option "Launch executable files without prompt" is checked.

For me it makes a perfect sense.  And should be pretty easy to implement ?!
```diff
index 918c31f..775764f 100644
--- a/src/filemenu.cpp
+++ b/src/filemenu.cpp
@@ -247,7 +247,7 @@ FileMenu::~FileMenu() {
 }
 
 void FileMenu::addTrustAction() {
-    if(info_->isExecutableType() && (!fileLauncher_ || !fileLauncher_->quickExec())) {
+    if(info_->isExecutableType() && !fileLauncher_) {
         QAction* trustAction = new QAction(files_.size() > 1
                                              ? tr("Trust selected executables")
                                              : tr("Trust this executable"),
```

**Comment by agaida** [comment]
@palinek - you are right, thats exactly what i will do for debian.

**Comment by tsujan** [comment]
> And should be pretty easy to implement

Wasn't about ease but enforcing a misconception.

**Comment by agaida** [comment]
No - it is not a misconception and never will be. So - if you don't like it - not my problem. The only misconception is written down in the current code. And please don't expect users to follow some obscure procedures.

The change that @palinek and myself suggest is perfectly fine, don't introduce any security flaws - so what.
Your misconception is easy to spot:
* One should be able to set/unset the trust all the time
* One might be able and free not to use the trust system (the don't ask hook in pcmanfm-qt's settings)
* Not using the trust system don't mean that i can't set/unset trust.

**Comment by tsujan** [comment]
> if you don't like it...

Frankly, I dislike double standards (https://github.com/lxqt/lxqt/issues/1661) more than letting an outdated option override a newer design.

**Comment by palinek** [comment]
> Wasn't about ease but enforcing a misconception.

I still don't follow the misconception. Why can't I (un)set the trust flag if I don't want the the double check for execution of executable/desktop files?
With the option "Launch executable files without prompt" checked I'm saying "I don't want to double check execution of executables/desktop" and not "I don't want to be able to set trust metadata to desktop files although .desktop files are visually presented as *(un)trusted*" IMO

**Comment by agaida** [comment]
Speaking of double standards - the solution is easy - lets remove the override. If you don't want that, make the override not wrecking functionality. Easy as can be.

**Comment by agaida** [comment]
@palinek - thats exactly my opinion

**Comment by tsujan** [comment]
>  Why can't I (un)set the trust flag if I don't want the the double check for execution of executable/desktop files?

The "trust" metadata has a history that's neglected here. The launching-without-prompt option was a security flaw, very inefficient and should have been removed. A file manager that allows it is like a file manager that allows copying folders into themselves: it isn't about "freedom" and its projection onto software ("I don't want... /  I want...").

However, removing a bad option suddenly wasn't a good idea. So, I made it as it is. The user sees what @agaida wrongly thinks is an issue (the misconception). Later, "Outdated" could be added in parenthesis. Yet later, it could be removed.

The above info can be found in previous discussion, on github and on reddit.

**Comment by agaida** [comment]
As i wrote - and you might don't like it too - approx. 80-85% of all linux desktops will not have this crippled functionality - ok, maybe arch and manjaro will have, but debian and all debian derivatives will work right.

**Comment by tsujan** [comment]
81-86% of Linux users may not want the "Root Instance" bar either -- bad argument ;)

**Comment by agaida** [comment]
it is not funny anymore, it leads to thoughts of leaving the project and abandon Linux at all on the desktop.

**Comment by agaida** [comment]
Debian and Siduction included - have better usage for my time. But hey, who cares.

**Comment by tsujan** [comment]
Overreaction after overreaction... Let's stop this! I do.

**Comment by agaida** [comment]
No, it is not a overreaction - it's the only logical consequence. I'm tired of fighting strawman arguments. And there are fine alternatives. Windows 10 become more usable each day. And with a debian integrated and lfs  it would fit my need for Linux to nearly 100%.

**Comment by agaida** [comment]
Let me be more verbose:
- it took me some man-days to prepare the releaese
- the release and all the things to do took one man-day
- preparing and uploading the release to debina two man days
- figthing about an option until now nearly a day - no, not funny

**Comment by tsujan** [comment]
> figthing about an option until now nearly a day - no, not funny

Making patches like https://github.com/lxqt/libfm-qt/pull/355 and testing them takes days -- another patch I've made for Apply button took several weeks; add to them life problems I have plenty of. Still, I have time to fight :D

**Comment by agaida** [comment]
https://bugs.launchpad.net/ubuntu/+source/pcmanfm-qt/+bug/1814326

**Comment by tsujan** [comment]
Quoting from there:

> Home, Root, and Temp folders have exclamation points over them

In plain English, it means they see `!` on those folder icons. If so, it isn't related to LXQt because the emblem is only for desktop files, not folders; if not, they speak a different language.
